mlir/lib/TableGen/Pattern.cpp

```diff
@@ -303,6 +303,12 @@ std::string SymbolInfoMap::SymbolInfo::getValueAndRangeUse(
   case Kind::Operand: {
     assert(index < 0);
     auto *operand = cast<NamedTypeConstraint *>(op->getArg(getArgIndex()));
+    if (operand->isOptional()) {
+      auto repl =
+          formatv(fmt, formatv("({0}.empty() ? Value() : *{0}.begin())", name));
```

Comment:
Could this use mlir::Value() instead of just Value()? Otherwise if the generated code is include from outside mlir namespace, it will not work. Or do we have a rule that all code that uses MLIR needs to be inside a mlir namespace?

Reply:
Nah there's no rule like that. If you send a PR with the proper namespacing I'll stamp/merge it (or I can send it tomorrow)

Reply:
I'll send one (elsehwere in this file we do fully namespace already)